### PR TITLE
Fix incomplete sorting.

### PR DESCRIPTION
**Pull request by eskimor**

### CODE REVIEW

File: runtime/parachains/src/disputes.rs [hunk 1]
```diff
@@ -130,7 +130,14 @@ where
 		(None, Some(_)) => Ordering::Greater,
 		(Some(_), None) => Ordering::Less,
 		// For local disputes, prioritize those that occur at an earlier height.
-		(Some(a_height), Some(b_height)) => a_height.cmp(&b_height),
+		(Some(a_height), Some(b_height)) => {
+			let height_ord = a_height.cmp(&b_height);
+			if height_ord == Ordering::Equal {
+				a.candidate_hash.cmp(&b.candidate_hash)
```
Comment (drahnr) [inline runtime/parachains/src/disputes.rs:136]: :+1:

**Review by drahnr — APPROVED** [review]

File: runtime/parachains/src/disputes.rs [hunk 2]
```diff
@@ -130,7 +130,14 @@ where
...
+			} else {
+				height_ord
+			}
```
Comment (bkchr) [inline runtime/parachains/src/disputes.rs:139]:
```suggestion
			a_height.cmp(&b_height).then_with(|| a.candidate_hash.cmp(&b.candidate_hash))
```

File: runtime/parachains/src/disputes.rs [hunk 3]
```diff
@@ -146,6 +147,34 @@ where
...
+	iter: I,
+	mut check_equal: C,
+) -> bool {
+	let mut iter = iter.into_iter();
+	if let Some(mut previous) = iter.next() {
+		while let Some(current) = iter.next() {
+			if check_equal(previous, current) {
+				return true
+			}
+			previous = current;
```
Comment (ordian) [inline runtime/parachains/src/disputes.rs:172]: Personally preferred the previous functional-style impl with windows(2). But this one is also readable, so no complaints.

**Review by ordian — APPROVED** [review]

File: runtime/parachains/src/disputes.rs (hunk 3 above)
Comment (drahnr) [reply runtime/parachains/src/disputes.rs:172]: Well, I prefer this one, it has no `unreachable!` or implicit unreachable, it's imho cleaner, yet a tad less idiomatic.

**Comment by drahnr** [comment]
bot merge

**Pull request merged**